genesis/recorders/base_recorder.py

```diff
@@ -148,25 +151,57 @@ def start(self):
         self._is_recording = True
 
         if self.run_in_thread:
-            self._data_queue = queue.Queue(maxsize=self._options.buffer_size)
-            self._processor_thread = threading.Thread(target=self._process_data_loop)
-            self._processor_thread.start()
+            self.start_thread()
 
     @gs.assert_built
     def stop(self):
         """Stop the recording thread and cleanup resources."""
         if self._is_recording:
             self._is_recording = False
-            self.sync()
+            if self.run_in_thread:
+                self.join_thread()
             self.cleanup()
 
     @gs.assert_built
-    def sync(self):
+    def join_thread(self):
         """Wait for the processor thread to finish."""
-        if self.run_in_thread and self._processor_thread is not None:
+        if self._processor_thread is not None:
             self._processor_thread.join()
             self._processor_thread = None
             self._data_queue = None
+        else:
+            gs.logger.warning(f"[{type(self).__name__}] join_thread(): No processor thread to join.")
+
+    @gs.assert_built
+    def start_thread(self):
+        """Wait for the processor thread to finish."""
+        if self._processor_thread is None:
+            self._data_queue = queue.Queue(maxsize=self._options.buffer_size)
+            self._processor_thread = threading.Thread(target=self._process_data_loop)
+            self._processor_thread.start()
+        else:
+            gs.logger.warning(f"[{type(self).__name__}] start_thread(): Processor thread already exists.")
+
+    @gs.assert_built
+    def sync(self, timestep: float = 0.1, timeout: float | None = None):
```

Comment:
Are you sure exposing timestep is relevant? It is usually never the case. I would use `timestep = min(0.1, timeout)`  for simplicity.